src/groups/power_group.jl

```diff
@@ -0,0 +1,342 @@
+
+Decorate a power manifold with a [`ProductOperation`](@ref).
+
+Constituent manifold of the power manifold must also have a [`IsGroupManifold`](@ref) or
+a decorated instance of one. This type is mostly useful for equipping the direct product of
+group manifolds with an [`Identity`](@ref) element.
+
+# Constructor
+
+    PowerGroup(manifold::AbstractPowerManifold)
+"""
+function PowerGroup(manifold::AbstractPowerManifold)
+    if !is_group_manifold(manifold.manifold)
+        error("All powered manifold must be or decorate a group.")
+    end
+    op = ProductOperation()
+    return GroupManifold(manifold, op)
+end
+
+@inline function active_traits(f, M::PowerGroup, args...)
+    if is_metric_function(f)
+        #pass to manifold by default - but keep Group Decorator for the retraction
+        return merge_traits(IsGroupManifold(M.op), IsExplicitDecorator())
+    else
+        return merge_traits(
+            IsGroupManifold(M.op),
+            active_traits(f, M.manifold, args...),
+            IsExplicitDecorator(),
+        )
+    end
+end
+
+function identity_element!(G::PowerGroup, p)
+    GM = G.manifold
+    N = GM.manifold
+    rep_size = representation_size(N)
+    for i in get_iterator(GM)
+        identity_element!(N, _write(GM, rep_size, p, i))
+    end
+    return p
+end
+function identity_element!(G::PowerGroupNestedReplacing, p)
+    GM = G.manifold
+    N = GM.manifold
+    for i in get_iterator(GM)
+        p[i...] = identity_element(N)
+    end
+    return p
+end
+
+function is_identity(G::PowerGroup, p::Identity{ProductOperation}; kwargs...)
+    return true
+end
+function is_identity(G::PowerGroup, p::Identity; kwargs...)
+    return false
+end
+function is_identity(G::PowerGroup, p; kwargs...)
+    GM = G.manifold
+    N = GM.manifold
+    rep_size = representation_size(N)
+    for i in get_iterator(GM)
+        if !is_identity(N, _read(GM, rep_size, p, i))
+            return false
+        end
+    end
+    return true
+end
+
+inv!(G::PowerGroup, q, ::Identity{ProductOperation}) = identity_element!(G, q)
+function inv!(G::PowerGroupNestedReplacing, q, ::Identity{ProductOperation})
+    return identity_element!(G, q)
+end
+function inv!(G::PowerGroup, q, p)
+    GM = G.manifold
+    N = GM.manifold
+    rep_size = representation_size(N)
+    for i in get_iterator(GM)
+        inv!(N, _write(GM, rep_size, q, i), _read(GM, rep_size, p, i))
+    end
+    return q
+end
+function inv!(G::PowerGroupNestedReplacing, q, p)
+    GM = G.manifold
+    N = GM.manifold
+    rep_size = representation_size(N)
+    for i in get_iterator(GM)
+        q[i...] = inv(N, _read(GM, rep_size, p, i))
+    end
+    return q
+end
+inv!(::PowerGroup, q::Identity{ProductOperation}, ::Identity{ProductOperation}) = q
+function inv!(
+    ::PowerGroupNestedReplacing,
+    q::Identity{ProductOperation},
+    ::Identity{ProductOperation},
+)
+    return q
+end
+
+_compose!(G::PowerGroup, x, p, q) = _compose!(G.manifold, x, p, q)
```

Comment:
`compose` here is dispatched on Level 2 not on level 1 – what is the idea here? For me they look like level 1.

Reply:
I copied it from the product group without thinking much. Probably the idea is that this way we don't need to handle `::Identity` disambiguiations.

Reply:
ah right, that sounds reasonable. Maybe we should add a comment to clarify?

Reply:
Sure.